packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,393 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { expect } from 'chai';
+
+import { BloomFilter } from '../../../src/remote/bloom_filter';
+import { ByteString } from '../../../src/util/byte_string';
+
+import {
+  testDataCount1Rate0001,
+  testDataCount1Rate01,
+  testDataCount1Rate1,
+  testDataCount50000Rate0001,
+  testDataCount50000Rate01,
+  testDataCount50000Rate1,
+  testDataCount5000Rate0001,
+  testDataCount5000Rate01,
+  testDataCount5000Rate1,
+  testDataCount500Rate0001,
+  testDataCount500Rate01,
+  testDataCount500Rate1,
+  testResultCount1Rate0001,
+  testResultCount1Rate01,
+  testResultCount1Rate1,
+  testResultCount50000Rate0001,
+  testResultCount50000Rate01,
+  testResultCount50000Rate1,
+  testResultCount5000Rate0001,
+  testResultCount5000Rate01,
+  testResultCount5000Rate1,
+  testResultCount500Rate0001,
+  testResultCount500Rate01,
+  testResultCount500Rate1
+} from './bloom_filter_golden_test_data';
+
+describe('BloomFilter', () => {
+  it('can initiate an empty bloom filter', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.size).to.equal(0);
+  });
+
+  it('empty bloom filter can have positive hash count', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 1);
+    expect(bloomFilter.size).to.equal(0);
+  });
+
+  it('should throw error if empty bloom filter inputs are invalid', () => {
+    try {
+      new BloomFilter(new Uint8Array(0), 1, 0);
```

Comment:
Use `expect(...).to.throw()` instead of using a try/catch block.

e.g. https://github.com/firebase/firebase-js-sdk/blob/fde5adf638dae4714ff7f25c75e475344907e05e/packages/firestore-compat/test/query.test.ts#L61

Reply:
I tried expect().to.throw(), but it is going to fail the test and completely ignores the "expect". 🤔

Reply:
Hmm. Okay, let's do some pair debugging on this.